[CI] Aggregation mixed cluster failure

A serialization failure, because of an unknown named writeable in `MixedClusterClientYamlTestSuiteIT test {p0=search.aggregation/330_auto_date_histogram/basic}`

Failure:

```
[2020-03-30T14:31:28,724][INFO ][o.e.b.MixedClusterClientYamlTestSuiteIT] [test] Stash dump on test failure [{
  "stash" : {
    "body" : {
      "error" : {
        "root_cause" : [
          {
            "type" : "illegal_argument_exception",
            "reason" : "Unknown NamedWriteable [org.elasticsearch.search.aggregations.PipelineAggregationBuilder][]",
            "stack_trace" : "org.elasticsearch.ElasticsearchException$1: Unknown NamedWriteable [org.elasticsearch.search.aggregations.PipelineAggregationBuilder][]
	at org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:644)
	at org.elasticsearch.action.search.SearchPhaseExecutionException.guessRootCauses(SearchPhaseExecutionException.java:167)
	at org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:626)
	at org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:572)
	at org.elasticsearch.rest.BytesRestResponse.build(BytesRestResponse.java:138)
	at org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:96)
	at org.elasticsearch.rest.BytesRestResponse.<init>(BytesRestResponse.java:91)
    ...
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: Unknown NamedWriteable [org.elasticsearch.search.aggregations.PipelineAggregationBuilder][]
	at org.elasticsearch.common.io.stream.NamedWriteableRegistry.getReader(NamedWriteableRegistry.java:112)
	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:45)
	at org.elasticsearch.common.io.stream.NamedWriteableAwareStreamInput.readNamedWriteable(NamedWriteableAwareStreamInput.java:39)
	at org.elasticsearch.search.aggregations.AggregatorFactories$Builder.<init>(AggregatorFactories.java:257)
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalWriteable(StreamInput.java:989)
	at org.elasticsearch.search.builder.SearchSourceBuilder.<init>(SearchSourceBuilder.java:204)
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalWriteable(StreamInput.java:989)
	at org.elasticsearch.search.internal.ShardSearchRequest.<init>(ShardSearchRequest.java:167)
	at org.elasticsearch.transport.RequestHandlerRegistry.newRequest(RequestHandlerRegistry.java:56)
	at org.elasticsearch.transport.InboundHandler.handleRequest(InboundHandler.java:187)
	at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:130)
	at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:107)
	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:681)
	at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:136)
	at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:110)
	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:68)
```

Build url: https://gradle-enterprise.elastic.co/s/wdpjhc2oh534q/tests/77udqutc4yt34-blndfefc2cef2

Reproduce line: `./gradlew ':qa:mixed-cluster:v7.8.0#mixedClusterTest' --tests "org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=search.aggregation/330_auto_date_histogram/basic}" -Dtests.seed=2DD4185E797F2251 -Dtests.security.manager=true -Dtests.locale=he -Dtests.timezone=Asia/Ujung_Pandang -Dcompiler.java=13`

This does not reproduce for me locally.

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/330_auto_date_histogram.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/search.aggregation/330_auto_date_histogram.yml
@@ -25,9 +25,6 @@ setup:
 
 ---
 "basic":
-  - skip:
-      version: " - 7.99.99"
-      reason: BWC test failures https://github.com/elastic/elasticsearch/issues/54396 && waiting for backport https://github.com/elastic/elasticsearch/pull/54379
   - do:
       search:
         body:

--- a/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java
+++ b/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java
@@ -127,7 +127,7 @@ public class AutoDateHistogramAggregationBuilder
     @Override
     protected void innerWriteTo(StreamOutput out) throws IOException {
         out.writeVInt(numBuckets);
-        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+        if (out.getVersion().onOrAfter(Version.V_7_3_0)) {
             out.writeOptionalString(minimumIntervalExpression);
         }
     }

--- a/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java
+++ b/server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java
@@ -110,20 +110,6 @@ public class AutoDateHistogramAggregationBuilder
 
     private String minimumIntervalExpression;
 
-    public String getMinimumIntervalExpression() {
-        return minimumIntervalExpression;
-    }
-
-    public AutoDateHistogramAggregationBuilder setMinimumIntervalExpression(String minimumIntervalExpression) {
-        if (minimumIntervalExpression != null && !ALLOWED_INTERVALS.containsValue(minimumIntervalExpression)) {
-            throw new IllegalArgumentException(MINIMUM_INTERVAL_FIELD.getPreferredName() +
-                " must be one of [" + ALLOWED_INTERVALS.values().toString() + "]");
-        }
-        this.minimumIntervalExpression = minimumIntervalExpression;
-        return this;
-    }
-
-
     /** Create a new builder with the given name. */
     public AutoDateHistogramAggregationBuilder(String name) {
         super(name);
@@ -135,6 +121,14 @@ public class AutoDateHistogramAggregationBuilder
         numBuckets = in.readVInt();
         if (in.getVersion().onOrAfter(Version.V_7_3_0)) {
             minimumIntervalExpression = in.readOptionalString();
+        }
+    }
+
+    @Override
+    protected void innerWriteTo(StreamOutput out) throws IOException {
+        out.writeVInt(numBuckets);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeOptionalString(minimumIntervalExpression);
         }
     }
 
@@ -157,16 +151,21 @@ public class AutoDateHistogramAggregationBuilder
     }
 
     @Override
-    protected void innerWriteTo(StreamOutput out) throws IOException {
-        out.writeVInt(numBuckets);
-        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
-            out.writeOptionalString(minimumIntervalExpression);
-        }
-    }
-
-    @Override
     public String getType() {
         return NAME;
+    }
+
+    public String getMinimumIntervalExpression() {
+        return minimumIntervalExpression;
+    }
+
+    public AutoDateHistogramAggregationBuilder setMinimumIntervalExpression(String minimumIntervalExpression) {
+        if (minimumIntervalExpression != null && !ALLOWED_INTERVALS.containsValue(minimumIntervalExpression)) {
+            throw new IllegalArgumentException(MINIMUM_INTERVAL_FIELD.getPreferredName() +
+                " must be one of [" + ALLOWED_INTERVALS.values().toString() + "]");
+        }
+        this.minimumIntervalExpression = minimumIntervalExpression;
+        return this;
     }
 
     public AutoDateHistogramAggregationBuilder setNumBuckets(int numBuckets) {

